src/Features/Core/Portable/Completion/CompletionHelper.cs

```diff
@@ -0,0 +1,536 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Globalization;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Completion.Providers;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Completion
+{
+    internal class CompletionHelper
+    {
+        private readonly CompletionRules _rules;
+        private readonly string _language;
+        private readonly CompletionService _completionService;
+
+        public CompletionHelper(CompletionRules rules, string language, CompletionService completionService)
+        {
+            _rules = rules;
+            _language = language;
+            _completionService = completionService;
+        }
+
+        public IReadOnlyList<TextSpan> GetHighlightedSpans(CompletionItem completionItem, string filterText)
+        {
+            var match = GetMatch(completionItem, filterText, includeMatchSpans: true);
+            return match?.MatchedSpans;
+        }
+
+        /// <summary>
+        /// Returns true if the completion item matches the filter text typed so far.  Returns 'true'
+        /// iff the completion item matches and should be included in the filtered completion
+        /// results, or false if it should not be.
+        /// </summary>
+        public bool MatchesFilterText(CompletionItem item, string filterText, CompletionTrigger trigger, CompletionFilterReason filterReason, ImmutableArray<string> recentItems = default(ImmutableArray<string>))
+        {
+            if (_language == LanguageNames.VisualBasic)
```

Comment:
Ew.... If necessary, can we do this through some sort of built in default logic, and then some sort of ILanguageService to allow langauges to customize?